protocols/imap/src/main/java/org/apache/james/imap/processor/ListProcessor.java

```diff
@@ -152,6 +153,11 @@ public char getHierarchyDelimiter() {
                     public MailboxPath getPath() {
                         return rootPath;
                     }
+
+                    @Override
+                    public MailboxId getId() {
+                        return null; //Will not be call in ListProcess scope
```

Comment:
Maybe you can even trhow an exception here no ? So we are really sure it's actually never called, and we can even avoid this comment